library/src/main/java/com/novoda/downloadmanager/lib/BatchRepository.java

```diff
@@ -3,66 +3,58 @@
+
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collection;
 import java.util.Collections;
 import java.util.List;
 
-import static com.novoda.downloadmanager.lib.Downloads.Impl.*;
-
 class BatchRepository {
 
     private static final List<Integer> PRIORITISED_STATUSES = Arrays.asList(
-            STATUS_CANCELED,
-            STATUS_RUNNING,
+            DownloadsStatus.STATUS_CANCELED,
+            DownloadsStatus.STATUS_RUNNING,
 
             // Paused statuses
-            STATUS_PAUSED_BY_APP,
-            STATUS_WAITING_TO_RETRY,
-            STATUS_WAITING_FOR_NETWORK,
-            STATUS_QUEUED_FOR_WIFI,
+            DownloadsStatus.STATUS_PAUSED_BY_APP,
+            DownloadsStatus.STATUS_WAITING_TO_RETRY,
+            DownloadsStatus.STATUS_WAITING_FOR_NETWORK,
+            DownloadsStatus.STATUS_QUEUED_FOR_WIFI,
 
-            STATUS_PENDING,
-            STATUS_SUCCESS
+            DownloadsStatus.STATUS_PENDING,
+            DownloadsStatus.STATUS_SUCCESS
     );
 
     private static final int PRIORITISED_STATUSES_SIZE = PRIORITISED_STATUSES.size();
 
-    private static final String[] PROJECT_BATCH_ID = {Batches._ID};
-    private static final String WHERE_DELETED_VALUE_IS = Batches.COLUMN_DELETED + " = ?";
+    private static final String[] PROJECT_BATCH_ID = {DownloadsColumnsBatches._ID};
+    private static final String WHERE_DELETED_VALUE_IS = DownloadsColumnsBatches.COLUMN_DELETED + " = ?";
     private static final String[] MARKED_FOR_DELETION = {"1"};
 
     private final ContentResolver resolver;
     private final DownloadDeleter downloadDeleter;
-    private final Uri batchContentUri;
-    private final Uri allDownloadsContentUri;
-
-    public static BatchRepository newInstance(ContentResolver resolver, DownloadDeleter downloadDeleter) {
-        return new BatchRepository(resolver, downloadDeleter, BATCH_CONTENT_URI, ALL_DOWNLOADS_CONTENT_URI);
-    }
+    private final DownloadsUriProvider downloadsUriProvider;
 
-    BatchRepository(ContentResolver resolver, DownloadDeleter downloadDeleter, Uri batchContentUri, Uri allDownloadsContentUri) {
+    BatchRepository(ContentResolver resolver, DownloadDeleter downloadDeleter, DownloadsUriProvider downloadsUriProvider) {
         this.resolver = resolver;
         this.downloadDeleter = downloadDeleter;
-        this.batchContentUri = batchContentUri;
-        this.allDownloadsContentUri = allDownloadsContentUri;
+        this.downloadsUriProvider = downloadsUriProvider;
     }
 
     void updateTotalSize(long batchId) {
         ContentValues updateValues = new ContentValues();
-        updateValues.put(Batches.COLUMN_TOTAL_BYTES, getSummedBatchSizeInBytes(batchId, COLUMN_TOTAL_BYTES));
-        resolver.update(batchContentUri, updateValues, Batches._ID + " = ?", new String[]{String.valueOf(batchId)});
+        updateValues.put(DownloadsColumnsBatches.COLUMN_TOTAL_BYTES, getSummedBatchSizeInBytes(batchId, DownloadsColumns.COLUMN_TOTAL_BYTES));
```

Comment:
Mixing columns from different column definitions is not good practise. Columns could be duplicated or column definitions subclassed.